src/test/e2e/nager-date-api.test.ts

```diff
@@ -0,0 +1,204 @@
+import request from "supertest"
+import { PUBLIC_HOLIDAYS_API_URL } from "../../config"
+
+const validContry = "GB"
+const invalidContry = "foo-bar"
+const validYear = 2024
+const invalidYear = "year"
+jest.setTimeout(300000)
+
+describe("Nager.Date API", () => {
+  describe("Country", () => {
+    describe("/CountryInfo", () => {
+      it("should return 200 and correct country info", async () => {
+        const { status, body } = await request(PUBLIC_HOLIDAYS_API_URL).get(
+          `/CountryInfo/${validContry}`
+        )
+
+        expect(status).toEqual(200)
+        expect(body).toEqual({
+          commonName: "United Kingdom",
+          officialName: expect.any(String),
+          countryCode: "GB",
+          region: "Europe",
+          borders: [expect.any(Object)],
+        })
+      })
+
+      it("should return 404 if the country is invalid", async () => {
+        const { status } = await request(PUBLIC_HOLIDAYS_API_URL).get(
+          `/CountryInfo/${invalidContry}`
+        )
+        expect(status).toEqual(404)
+      })
+    })
+    describe("/AvailableCountries", () => {
+      it("should return 200 and correct country info", async () => {
+        const { status, body } = await request(PUBLIC_HOLIDAYS_API_URL).get(
+          "/AvailableCountries"
+        )
+
+        expect(status).toEqual(200)
+        expect(body).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              countryCode: expect.any(String),
+              name: expect.any(String),
+            }),
+          ])
+        )
+      })
+    })
+  })
+
+  describe("LongWeekend", () => {
+    describe("/LongWeekend", () => {
+      it("should return 200 and correct info for the long weekends", async () => {
+        const { status, body } = await request(PUBLIC_HOLIDAYS_API_URL).get(
+          `/LongWeekend/${validYear}/${validContry}`
+        )
+        expect(status).toEqual(200)
+        expect(body).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              startDate: expect.any(String),
+              endDate: expect.any(String),
+              dayCount: expect.any(Number),
+              needBridgeDay: expect.any(Boolean),
+            }),
+          ])
```

Comment:
It is a good way to assert the content of the response, but I would suggest to use snapshots to avoid complex assertion and keep it easy to track any changes:
`expect(body).toMatchSnapshot();`
https://jestjs.io/docs/snapshot-testing

Reply:
this was also a recommendation

Reply:
I like this suggestion because I mostly did it to experiment with some JEST functions and I didn't know this one existed, but I really like it, so I'm going to toy around with it!